server/router.js

```diff
@@ -47,68 +45,58 @@ module.exports = function (opts, db) {
     })
   }
 
-  router.on('/view/:archiveKey', {
-    get: function (req, res, params) {
-      archiveRoute(params.archiveKey, function (state) {
-        return sendSPA('/view/:archiveKey', req, res, params, state)
-      })
-    }
+  router.get('/view/:archiveKey', function (req, res) {
+    archiveRoute(req.params.archiveKey, function (state) {
+      return sendSPA('/view/:archiveKey', req, res, state)
+    })
   })
 
-  router.on('/:username/:dataset', {
-    get: function (req, res, params) {
-      // TOOD: do it in one db query not two
-      db.models.users.get({username: params.username}, function (err, user) {
+  router.get('/:username/:dataset', function (req, res) {
+    // TOOD: do it in one db query not two
+    db.models.users.get({username: req.params.username}, function (err, user) {
+      if (err) return res.end(err.message)
+      if (!user.id) return sendSPA('/404', req, res)
```

Comment:
This needs to be `var user = results[0]` or something since query will return array. 

Forgot to fix this last night.